src/dtype.rs

```diff
@@ -576,13 +561,39 @@ impl<'py> PyArrayDescrMethods<'py> for Bound<'py, PyArrayDescr> {
         unsafe { PyType::from_borrowed_type_ptr(self.py(), dtype_type_ptr) }
     }
 
+    fn itemsize(&self) -> usize {
+        unsafe { PyDataType_ELSIZE(self.py(), self.as_dtype_ptr()).max(0) as _ }
+    }
+
+    fn alignment(&self) -> usize {
+        unsafe { PyDataType_ALIGNMENT(self.py(), self.as_dtype_ptr()).max(0) as _ }
+    }
+
+    fn flags(&self) -> u64 {
+        unsafe { PyDataType_FLAGS(self.py(), self.as_dtype_ptr()) as _ }
+    }
+
+    fn ndim(&self) -> usize {
+        if !self.has_subarray() {
+            return 0;
+        }
+        unsafe {
+            PyTuple_Size((*PyDataType_SUBARRAY(self.py(), self.as_dtype_ptr())).shape).max(0) as _
+        }
+    }
+
     fn base(&self) -> Bound<'py, PyArrayDescr> {
         if !self.has_subarray() {
```

Comment:
Now that we need those accessor functions, I think it would be nice to use only one access, i.e. something like

```rust
let subarray = unsafe { PyDataType_SUBARRAY(self.py(), self.as_dtype_ptr()).as_ref() };
if let Some(subarray) = subarray {
  unsafe {
    Bound::from_borrowed_ptr(self.py(), subarray.base.cast()).downcast_into_unchecked()
  }
} else {
  return 0;
}
```

and similar in the other places below.

Reply:
Done in #442.